### PR TITLE
implement random transformation

### PR DESCRIPTION
**Pull request by PaulBautin**
To mimic subject repositioning, for each rescaling, it would be nice to have a random translation and rotation of the image. This method generates randomized affine transformed images, one per subject before rescaling.

DONE:
- create affine-transfo.py
- generate gaussian distribution random values
- rotate and shift image in every direction on all subjects

### CODE REVIEW

File: affine_transfo.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+# implement random transformations to mimic subject repositioning, for each rescaling,
+#
+# generate randomized transfo per subject, and freeze them in the repos,
+# so that we can reproduce the results by inputting the frozen params in subsequent runs of the pipeline,
+# shifting (±5 voxels in each direction),
+# rotation (±10° in each direction),
+#
+# ---------------------------------------------------------------------------------------
+# Authors: Paul Bautin
+# inspired from
```
Comment (jcohenadad) [inline affine_transfo.py:14]: add source

File: affine_transfo.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+#
+# About the license: see the file LICENSE.TXT
```
Comment (jcohenadad) [inline affine_transfo.py:16]:
```suggestion
# About the license: see the file LICENSE
```

File: affine_transfo.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+from skimage.transform import rotate
+
+import nibabel as nib
+
+from  scipy.ndimage import shift
+
+
+##############################VALUES###############################
+  # generate Gaussian values
```
Comment (jcohenadad) [inline affine_transfo.py:32]:
```suggestion
  # generate Gaussian-distributed random values
```

File: affine_transfo.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+     increase_axes = np.sqrt(2*increase_angle**2)
+     min_pad = math.ceil(increase_axes+np.sqrt(3*(max_shift**2)))
+     # padding choices: {‘constant’, ‘edge’, ‘symmetric’, ‘reflect’, ‘wrap’}
+     data = np.pad(data, min_pad, 'constant')
+     print('min padding:',min_pad)
+     print('data shape (image with padding):',data.shape)
+     return data, min_pad
+
+
+  ##############################ROTATION###############################
```
Comment (jcohenadad) [inline affine_transfo.py:82]: the function seems to deal with both rotation and translation, so this header is confusing. In general, i would get rid of those headers-- this is double work to maintain, and this is what the docstrings are for.

if you use an IDE, the docstrings appear with another formatting, so they are easy to visually spot

File: affine_transfo.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+ def transfo(angle_IS, angle_AP, angle_RL, shift_LR, shift_PA, shift_IS, data):
+     """apply rotation and translation on image
+     :param data: padded image data
```
Comment (jcohenadad) [inline affine_transfo.py:85]: doesn't seem to be the only input

**Comment by PaulBautin** [comment]
Requested corrections have been addressed. For clarity I will soon merge this branch as the addition of "affine_transfo.py" can be reviewed directly in branch "master".

**Pull request merged**